src/engine/threaded_engine.cc

```diff
@@ -333,9 +333,11 @@ void ThreadedEngine::PushAsync(AsyncFn fn, Context exec_ctx,
         << device_count_;
   }
 #endif
-  ThreadedOpr *opr = NewOperator(std::move(fn), const_vars, mutable_vars, prop, opr_name, wait);
-  opr->temporary = true;
   const bool profiling = profiler_->IsProfiling(profiler::Profiler::kImperative);
+	const char* d_name = profiling ? profiler::CustomOpProfiler::Get()->GetDisplayName(opr_name) : NULL;
+	const char* display_name = d_name ? d_name : opr_name;
```

Comment:
Why use a std::string if just const char*?

Reply:
I would still use std::string to keep consistent with rest of codebase. GetDisplayName should just return std::string, They are also just more maintainable and flexible.

Reply:
There is a good reason for using chat*. Because display_name is passed to the next function (which takes a char*), we would want to avoid using a local variable here so that we are not pointing to destrcuted data in the future.

Reply:
The Idea is GetDisplayName() will store the display name in the singleton object, and return a char* to that data. I need to change the function name to something like "AssignDisplayName" though to avoid confusion